src/pkg/projects/dir.targets

```diff
@@ -158,6 +158,14 @@
       <Output TaskParameter="TargetOutputs" ItemName="SharedFrameworkRuntimeFiles" />
     </MSBuild>
 
+    <!--
+      Workaround: zero-versioned Microsoft.VisualBasic.dll in non-Windows CoreFX transport package.
```

Comment:
I was actually thinking you would constrain to a single filename.

Reply:
That would be better, but I'm not sure if the risk of another non-Windows file changing to 0.0.0.0/none before https://github.com/dotnet/corefx/issues/36630 is fixed is high enough to warrant effort here. This is also a backstop for a brand new source-build-only fix that we stumbled into (for a bug nobody has mentioned before) rather than actually preventing a regression.

Not hard to do, but this was a bit quicker and I wanted to get this out of the way.

Reply:
I see.  One thing I've thought about in the past is constructing the platform manfiest from some form of authoring rather than inspecting binaries.  That way we just inspect the binaries in the individual legs to validate that they are represented correctly in the manifest, but the manifest represents a superset of all legs.  When maestro picks up an update it runs a target that commits an updated manifest.

Reply:
Also, it might make sense to do more "layout correctness" checks in a central place.  Things like this and verify closure are relevant for more things than just the places where we run them now.

Reply:
Yeah, all this stuff happening recently has been making me think about how core-setup could really use general artifact testing, rather than trying to add little checks to various individual parts of the build.

> When maestro picks up an update it runs a target that commits an updated manifest.

Seems like this would make the build somewhat faster, too, since we wouldn't have to download all platforms' runtime packages on all legs to generate the platform manifest.

Unfortunately not supported in Maestro++ unless it's coded into Darc directly. https://github.com/dotnet/arcade/issues/774. I think it'd have to be pretty general and robust because that limits our opportunities to change/maintain it quite a bit. I think there's also expectations of "fast" and not requiring a real repo clone and build tooling. I can only imagine it being reasonable if everything needed is in BAR metadata.

I wondered a bit about the style of baseline + update baseline target + PR validation build letting you download an updated baseline (we use this in source-build). Problem is the versions change for every single update--so I don't think that would work at all.